serviceinfo-api-server/src/main.rs

```diff
@@ -224,7 +227,64 @@ async fn serviceinfo_handler(
         .modules
         .contains(&FedoraIotServiceInfoModule::SSHKey.into())
     {
-        if let Some(initial_user) = &user_data.service_info_configuration.settings.initial_user {
+        //here we first check if the reuested device has device-specific file stored
+        //in device_specific_store_driver ,if not this device gets config from base file
+        let settings: ServiceInfoApiServerSettings = match settings_for("serviceinfo-api-server") {
+            Ok(config) => config.try_deserialize(),
+            Err(e) => {
+                return Err(warp::reject::custom(RtrFailure(e)));
+            }
+        }
+        .map_err(|e| warp::reject::custom(RtrFailure(e.into())))?;
+
+        let path_per_device_store = match settings.device_specific_store_driver {
+            StoreConfig::Directory { mut path } => {
+                let file_name = format!(
+                    "{}.yml",
+                    query_info.device_guid.to_string().replace('\"', "")
+                );
+                path.push(file_name);
+                path.to_string_lossy().into_owned()
+            }
```

Comment:
I'd mask all of this behind the store no? ultimately, we do want to do `device_specific_store.load_data(device_uuid)`

Reply:
Agreed

Reply:
I ultimately think this has to be "abstracted" away so that it's easier in the future to switch between serialization/implementation/whatever

Reply:
I agree, later on when we consider to override all of the service info (rather than just sshkey &username) the above part is going to be common. So as a part of abstraction I was thinking of moving that to `settings_per_device ` function.

Reply:
So as discussed , added some level of abstraction to this function, 
  and later when we have Database addition we will consider changes accordingly.